### DISCUSSION TITLE
GoogleIDFASupport should be supported in auto-linking

### DISCUSSION THREAD

**Discussion by hakkikonu** [invertase/react-native-firebase#4393] [Ideas] [answered]
On this doc page https://rnfirebase.io/analytics/usage/installation/ios#device-identification

There is a pod line `GoogleIDFASupport` and currently it's not included in auto-linking.
This is necessary for device identification according to Google Docs: https://support.google.com/firebase/answer/6318039

If you're using Google Ads for your app and if you don't have `GoogleIDFASupport`, your conversions will be  '0' in Google Ads reports without this package on IOS side

**Comment by mikehardy** [comment, marked as answer]
We will not auto-link that in, including (or not including) IDFA is a big decision every app needs to make on it's own. Imagine if we auto-linked that in for a kids app for instance...

**Reply by Bardiamist** [reply]
Hey @mikehardy 
Is `GoogleIDFASupport` still required? Looks its very old and not ready for arm64 and it's not mentioned [here](https://firebase.google.com/support/guides/analytics-adsupport?sjid=14143299795842744466-EU).

**Reply by mikehardy** [reply]
> Hey @mikehardy 
> Is `GoogleIDFASupport` still required? Looks its very old and not ready for arm64 and it's not mentioned [here](https://firebase.google.com/support/guides/analytics-adsupport?sjid=14143299795842744466-EU).

I haven't had arm64 problems with it. It's not required as you don't have to link it but I just investigated this in response to a question on a related issue and while the docs are subtle my analysis is that if you want the extra analytics you still link it. If you have a kids app or similar you don't link it *and* you use the analytics dependency that doesn't even have the symbols for injectable library presence

**Reply by Bardiamist** [reply]
@mikehardy 

>I haven't had arm64 problems with it

I use M1.
It's ok when build on the real device but when build on simulator I have:

```
Building for 'iOS-simulator', but linking in object file (/Users/bardiamist/Documents/proj/proj-mobile/ios/Pods/GoogleIDFASupport/Libraries/libAdIdAccessLibrary.a[arm64][2](TAGActualAdIdAccess.o)) built for 'iOS'
```

Still not sure that `GoogleIDFASupport` required for me. I'd want to provide best analytics expirience, use IDFA. I linked AdSupport.framework. Will AdSupport.framework work without GoogleIDFASupport?

Where can I found documentation for what GoogleIDFASupport required?

**Reply by mikehardy** [reply]
Most recent analysis https://github.com/invertase/react-native-firebase/issues/7027#issuecomment-1908481739
